app/models/user.rb

```diff
@@ -101,9 +101,12 @@ def with_team_kind(kind)
     end
 
     def with_all_associations_joined
-      includes(:conferences).group("conferences.id").
-      includes(:roles).group("roles.id").
-      includes(roles: :team).group("teams.id")
+      eager_load(:conferences).
```

Comment:
Using `eager_load` ensures that any associations are loaded using a single query (per association). For example: a `User` can have many `Post` objects. Lets say you then get a list of users and do something with their posts:

```ruby
User.all.each do |user|
  user.posts.each do |post|

  end
end
```

Because associations are (by default) only loaded when needed this will result in 1 query for all users and 1 query for every user to get their posts. Thus, if you have 10 users you'll end up with 1+10 queries.

Using `eager_load` prevents this from happening by loading all `Post` objects using a single query, meaning the above snippet only needs 2 queries.